### PR TITLE
refactor(payment): STRIPE-423 StripeV3 moved to separate package

### PR DESCRIPTION
**Pull request by bc-dronov**
## What?
StripeV3 moved to separate package

## Testing / Proof
Manual tests/ unit tests
@bigcommerce/team-checkout

### CODE REVIEW

File: packages/stripe-integration/.eslintrc.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
       "rules": {
         "@typescript-eslint/no-unsafe-argument": "off",
         "@typescript-eslint/no-unsafe-assignment": "off",
-        "@typescript-eslint/no-unsafe-member-access": "off"
+        "@typescript-eslint/no-unsafe-member-access": "off",
+        "@typescript-eslint/consistent-type-assertions": "off"
```
Comment (PavlenkoM) [inline packages/stripe-integration/.eslintrc.json:10]: if we need this rule only for test, will be better to keep this rule only for test files
```
{
      "files": ["*.test.tsx"],
      "rules": {
          "@typescript-eslint/consistent-type-assertions": "off"
        }
}
```

File: packages/stripe-integration/src/stripev3/StripePaymentMethod.spec.tsx [hunk 2]
```diff
@@ -1,59 +1,37 @@
+/* eslint @typescript-eslint/no-floating-promises: 0 */
```
Comment (PavlenkoM) [inline packages/stripe-integration/src/stripev3/StripePaymentMethod.spec.tsx:1]: we should migrate old `enzyme` test to `@testing-library/react`. But you can do this in separate PR. Create please a task for this, and for Stripe UPE as well.

**Review by PavlenkoM — APPROVED** [review]

File: packages/stripe-integration/src/stripev3/StripePaymentMethod.spec.tsx (hunk 2 above)
Comment (vitalii-koshovyi) [reply packages/stripe-integration/src/stripev3/StripePaymentMethod.spec.tsx:1]: If you don't have problems with deadlines, I would suggest to implement it in this PR.

Comment (bc-dronov) [reply packages/stripe-integration/src/stripev3/StripePaymentMethod.spec.tsx:1]: No problem, I can do it in this PR

Comment (bc-dronov) [reply packages/stripe-integration/src/stripev3/StripePaymentMethod.spec.tsx:1]: @vitalii-koshovyi If you don't mind, I'll create a separate task for this.

**Review by bc-peng — APPROVED** [review]
LGTM.
